t3/connection/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+            .max_send_message_len(std::i32::MAX)
+            .connect_to_uri(uri, &logger);
+
+        let transaction_service_client = TransactionServiceClient::new(ch);
+
+        Self {
+            uri: uri.clone(),
+            api_key,
+            transaction_service_client,
+            logger,
+        }
+    }
+
+    pub fn find_transactions(&self) -> Result<FindTransactionsResponse, Error> {
+        let request = FindTransactionsRequest::new();
+
+        Ok(self
+            .transaction_service_client
+            .find_transactions_opt(&request, common_headers_call_option(&self.api_key))
+            .map_err(|err| {
+                log::warn!(
+                    self.logger,
+                    "t3_transaction_service find_transactions RPC call failed: {}",
+                    err
+                );
+                err
+            })?)
+    }
+
+    pub fn list_transactions() {}
+
+    pub fn create_transaction(
+        &self,
+        transparent_transaction: TransparentTransaction,
+    ) -> Result<CreateTransactionResponse, Error> {
+        let mut request = CreateTransactionRequest::new();
+        request.set_transaction(transparent_transaction);
+
+        Ok(self
+            .transaction_service_client
+            .create_transaction_opt(&request, common_headers_call_option(&self.api_key))
+            .map_err(|err| {
+                log::warn!(
+                    self.logger,
+                    "t3_transaction_service find_transactions RPC call failed: {}",
+                    err
+                );
+                err
+            })?)
+    }
+
+    pub fn test_error(&self, code: i32) -> Result<TestErrorResponse, Error> {
+        let mut request = TestErrorRequest::new();
+        request.set_code(code);
+
+        Ok(self
+            .transaction_service_client
+            .test_error_opt(&request, common_headers_call_option(&self.api_key))
+            .map_err(|err| {
+                log::warn!(
+                    self.logger,
+                    "t3_transaction_service find_transactions RPC call failed: {}",
+                    err
+                );
+                err
+            })?)
+    }
+}
+
+impl Display for T3Connection {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        write!(f, "{}", self.uri)
+    }
+}
+
+impl Eq for T3Connection {}
+
+impl Hash for T3Connection {
+    fn hash<H: Hasher>(&self, hasher: &mut H) {
+        self.uri.addr().hash(hasher);
+    }
+}
+
+impl PartialEq for T3Connection {
+    fn eq(&self, other: &Self) -> bool {
+        self.uri.addr() == other.uri.addr()
+    }
+}
+
+impl Ord for T3Connection {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.uri.addr().cmp(&other.uri.addr())
+    }
+}
+
+impl PartialOrd for T3Connection {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.uri.addr().partial_cmp(&other.uri.addr())
+    }
+}
+
+impl Connection for T3Connection {
```

Comment:
Do you actually need this?

Reply:
This seems to be the standard with the other Connection types found throughout the mobilecoin repo

Reply:
I see, okay.